### PR TITLE
Add recipe for napari-console

### PR DESCRIPTION
**Pull request by jni**
This PR adds a recipe for napari-console, a plugin for napari. napari has until now included a built-in IPython console. We are now pulling it out and making it a plugin instead, which is implemented in this package.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] ~~If static libraries are linked in, the license of the static library is packaged.~~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sofroniewn** [comment]
I agree to be listed as a maintainer! Thanks!!

**Comment by tlambert03** [comment]
I agree. Thanks!

**Comment by jni** [comment]
Hmm, I'm kinda confused here about why we need setuptools_scm (the setup.py call appears to be failing because of it) when installing from the sdist... @tlambert03 do you know why that would be?

**Comment by tlambert03** [comment]
that's what generates the `_version.py` file from the git tags at build time.  (`_version.py` isn't checked into source).  We [have it in napari too](https://github.com/conda-forge/napari-feedstock/blob/6ac706289ff075b1dce7ba48b12dc4f9d76fbb56/recipe/meta.yaml#L22)

**Comment by jni** [comment]
Does this mean the source needs to be github rather than pypi? pypi won't have any of the tags, right?

**Comment by tlambert03** [comment]
no... the source is pypi for napari too, I'm not immediately sure... need to look at the build logs

**Comment by jni** [comment]
Bizarre... Ok I'll start by adding setuptools_scm to the build reqs... 🙏

**Comment by jni** [comment]
Hmmmm. Any guesses @tlambert03? My understanding is that `_version.py` should already be in the sdist and so we shouldn't need setuptools_scm...

**Comment by tlambert03** [comment]
yeah, I agree, I'm confused too... but for now, add it to host as well, like in the napari recipe

**Comment by tlambert03** [comment]
well, farther now... but failing because it looks like napari-console [depends on napari](https://github.com/napari/napari-console/blob/main/napari_console/qt_console.py#L14), though it's not listed in requirements.txt

**Comment by sofroniewn** [comment]
> well, farther now... but failing because it looks like napari-console depends on napari, though it's not listed in requirements.txt

Opps! My bad! I forgot about that line for the theme/ stylesheets stuff (hopefully one day it will go away!). `napari` is a test dependency and so I missed that. Should I add napari as a dependency and make `0.0.2` release or are we going to get into some circular dependency weirdness?

**Comment by jni** [comment]
> Should I add napari as a dependency and make 0.0.2 release or are we going to get into some circular dependency weirdness?

@Carreau do you have any thoughts on this?

**Comment by sofroniewn** [comment]
I've made a PR here https://github.com/napari/napari-console/pull/9 to add the napari dependency, we can merge that and make a `0.0.2` napari-console release and try this again. not sure if we'll also need to do something about pyside2 / pyqt5 more explicitly too

File: recipes/napari-console/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{% set name = "napari-console" %}
+{% set version = "0.0.1" %}
```
Comment (sofroniewn) [inline recipes/napari-console/meta.yaml:2]:
```suggestion
{% set version = "0.0.2" %}
```

Comment (sofroniewn) [reply recipes/napari-console/meta.yaml:2]: @jni can you update this when you get the chance

File: recipes/napari-console/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: e30e06b44cbe871ac73dd939eb786a18ee7ce367d599cc5e3724c0610dc73838
```
Comment (tlambert03) [inline recipes/napari-console/meta.yaml:10]:
```suggestion
  sha256: 3fc9370bc15cb156c1840c7522045a18eed3cc85d660f0d36ca6e421a9caaa3e

```

**Comment by jni** [comment]
🤞

**Comment by jni** [comment]
Who knew such a tiny package could cause so many problems 😂 

Ok so now the issue is

```pytb
  File "/home/conda/staged-recipes/build_artifacts/napari-console_1611874859084/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.9/site-packages/qtpy/QtGui.py", line 21, in <module>
    from PyQt5.QtGui import *
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

I can't remember how we dealt with such things in napari... I'll have a look.

**Comment by jni** [comment]
Hmmm, I went to look for the [qtconsole recipe](https://github.com/conda-forge/qtconsole-feedstock/blob/master/recipe/meta.yaml) to see how they solve this (since this is in that import), and I don't see an immediate fix... Unless [this line](https://github.com/conda-forge/qtconsole-feedstock/blob/35e5d553ce5b79bbd62bbbc184c87bf90ade3f4a/recipe/meta.yaml#L35) somehow alters the state of the machine?

**Comment by jni** [comment]
Ah ha! I *think* this might be it?

https://github.com/conda-forge/qtconsole-feedstock/blob/master/recipe/yum_requirements.txt

**Comment by jni** [comment]
@conda-forge/help-python this is ready for review/merge! 🎉 🙏

**Review by sofroniewn — APPROVED** [review]
LGTM! Nice work all, I think it's good to merge

**Comment by jni** [comment]
@conda-forge/help-python is anyone available to merge this? 🙏

**Comment by sofroniewn** [comment]
@conda-forge/staged-recipes this is ready for review! Thanks!!

**Pull request merged**